src/accelerate/utils/dataclasses.py

```diff
@@ -986,11 +987,36 @@ def set_auto_wrap_policy(self, model):
                     else:
                         transformer_cls_to_wrap.add(transformer_cls)
 
-                self.auto_wrap_policy = functools.partial(
+                auto_wrap_policy = functools.partial(
                     transformer_auto_wrap_policy,
                     # Transformer layer class to wrap
                     transformer_layer_cls=transformer_cls_to_wrap,
                 )
+
+                # In an FSDP setting PEFT models require individually wrapping unfrozen parameters
+
+                if isinstance(model, PeftModel):
+                    print("PEFT wrapping")
+                    from torch.distributed.fsdp.wrap import (
+                        _or_policy,
+                        lambda_auto_wrap_policy,
+                        transformer_auto_wrap_policy,
+                    )
+
+                    def lambda_policy_fn(module):
+                        if (
+                            len(list(module.named_children())) == 0
```

Comment:
I assume that this is called recursively on all sub-modules of the model, and you're checking here that they are "leaf nodes". One thing I wonder: Is it enough to only check for `weight`? What if the learnable parameter has a different name, would this fail? Maybe it's better to check `and any(p.requires_grad for p in module.parameters()`?

Reply:
Good point. I just ran some tests with different kind of models and your suggestion works flawlessly - i therefore changed it.